pkg/irel/root.go

```diff
@@ -16,12 +16,71 @@
 
 package irel
 
-import "github.com/spf13/cobra"
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
 
-var Root = &cobra.Command{
-	Use:               "irel",
-	Short:             "irel is a tool for relocating container images",
-	Run:               func(cmd *cobra.Command, _ []string) { cmd.Usage() },
-	DisableAutoGenTag: true,
+	"github.com/pivotal/image-relocation/pkg/registry"
+	"github.com/pivotal/image-relocation/pkg/registry/ggcr"
+	"github.com/spf13/cobra"
+)
+
+var (
+	caCertPaths   []string
+	skipTLSVerify bool
+
+	Root = &cobra.Command{
+		Use:               "irel",
+		Short:             "irel is a tool for relocating container images",
+		Run:               func(cmd *cobra.Command, _ []string) { cmd.Usage() },
+		DisableAutoGenTag: true,
+	}
+)
+
+func init() {
+	Root.PersistentFlags().StringSliceVarP(&caCertPaths, "ca-cert-path", "", nil, "Path to CA certificate used to verify TLS certificates for registries (can be specified more than once)")
+	Root.PersistentFlags().BoolVarP(&skipTLSVerify, "skip-tls-verify", "", false, "Skip TLS certificate verification for registries")
 }
 
+func mustGetRegistryClient() registry.Client {
+	transport, err := httpTransport(caCertPaths, skipTLSVerify)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return ggcr.NewRegistryClient(ggcr.WithTransport(transport))
+}
+
+func httpTransport(certs []string, insecureSkipVerify bool) (http.RoundTripper, error) {
```

Comment:
This is sufficiently involved that I think it should move to a `transport` package and have unit tests added. May be best to return `*http.Transport` rather than `http.RoundTripper` to facilitate easy testing.